glam/api/views.py

```diff
@@ -690,21 +691,88 @@ def _get_random_probes(data_source, random_percentage, limit):
         )
     return aggs
 
+def _get_fx_most_used_probes(days=30, limit=9):
+    """Retrieve most used probes from BQ
 
-@api_view(["POST"])
+    :param int days: The amount of days to consider for recency
+    :param int limit: The max amount of probes to retrieve
+    """
+
+    dimensions = []
+    today = datetime.now()
+    min_date = today - timedelta(days=days)
+    dimensions.append(Q(timestamp__gte=min_date))
+    dimensions.append(Q(timestamp__lte=today))
+
+    result = UsageInstrumentation.objects.filter(*dimensions)
+    most_used_probes = result.values("probe_name").annotate(total=Count("*")).order_by("-total",)
+    probe_names = [f'"{p["probe_name"]}"' for p in most_used_probes]
+
+    legacy_table_name = (f"`{GLAM_BQ_PROD_PROJECT}.glam_etl.glam_desktop_nightly_aggregates`")
+    fog_table_name = (f"`{GLAM_BQ_PROD_PROJECT}.glam_etl.glam_fog_nightly_aggregates`")
+
+    query = f"""
+        WITH fx_metrics AS (
+            SELECT * EXCEPT(
+                    non_norm_histogram,
+                    non_norm_percentiles)
+                FROM
+                    {legacy_table_name}
+            UNION ALL (
+                SELECT * EXCEPT(
+                    app_id,
+                    channel,
+                    ping_type)
+                FROM
+                    {fog_table_name})),
+        selected_version AS (
+            SELECT
+                ARRAY_AGG(DISTINCT version
+                    ORDER BY version DESC
+                    LIMIT 2)[SAFE_OFFSET(1)] # Second most recent version has more data
+                AS v
+            FROM
+                fx_metrics
+            WHERE
+                version < 1000) # Avoids the 1024 version
+        SELECT
+            distinct metric,
+            ARRAY_AGG(histogram)[SAFE_OFFSET(0)] AS histogram
+        FROM
+            fx_metrics,
+            selected_version
+        WHERE
+            build_id='*'
+            AND os='*'
+            AND metric_key=''
+            AND metric_type NOT IN ('boolean',
+                'histogram-boolean',
+                'scalar')
+            AND metric IN ({", ".join(probe_names)})
+            AND version = selected_version.v
+        GROUP BY metric
+        LIMIT {limit}"""
+
+    with bigquery.Client() as client:
+        aggs = client.query(query)
+    return aggs
+
+
+@api_view(["GET"])
```

Comment:
GET is the right method for this. Besides, django caching only works with HEAD or GET.